### PR TITLE
valid SSID list and vendor tag for Wireless discovery

### PR DESCRIPTION
**Pull request by Keksgesicht**
User Stories:
- BP-18 | collect hostapd instaces based on ssid
- BP-21 | IPv6 Address in Vendor Element

### CODE REVIEW

File: ubus.c [hunk 1]
```diff
@@ -270,6 +272,68 @@ usteer_ubus_set_config(struct ubus_context *ctx, struct ubus_object *obj,
...
+		char *ssid = blobmsg_data(cur);
+		uss = malloc(sizeof(struct usteer_ssid));
+		strncpy(uss->ssid, ssid, sizeof(uss->ssid));
+		list_add(&uss->list, config_ssid_list);
+	}
+
+	usteer_local_nodes_init(ubus_ctx);
+}
+
+void config_get_ssid(struct blob_buf *buf) {
```
Comment (NeXX451) [inline ubus.c]: void config_get_ssid(struct blob_buf *buf)
 {

File: ubus.c [hunk 2]
```diff
@@ -270,6 +272,68 @@ usteer_ubus_set_config(struct ubus_context *ctx, struct ubus_object *obj,
 	return 0;
 }
 
+struct usteer_ssid {
+	char ssid[33];
+	struct list_head list;
+};
+static struct list_head *config_ssid_list = NULL;
+
+void config_set_ssid(struct blob_attr *data) {
```
Comment (NeXX451) [inline ubus.c]: void config_set_ssid(struct blob_attr *data)
 {

File: ubus.c [hunk 3]
```diff
@@ -270,6 +272,68 @@ usteer_ubus_set_config(struct ubus_context *ctx, struct ubus_object *obj,
...
+	void *c;
+
+	c = blobmsg_open_array(buf, "ssid");
+	list_for_each_entry(uss, config_ssid_list, list) {
+		blobmsg_add_string(buf, NULL, uss->ssid);
+	}
+	blobmsg_close_array(buf, c);
+}
+
+bool usteer_is_valid_ssid(const char *ssid) {
```
Comment (NeXX451) [inline ubus.c]: bool usteer_is_valid_ssid(const char *ssid)
 {

**Review by NeXX451 — CHANGES_REQUESTED** [review]
fix braces

**Review by Peter-Nicomann — APPROVED** [review]

**Review by NeXX451 — APPROVED** [review]

**Pull request merged**